components/Login.js

```diff
@@ -0,0 +1,92 @@
+import { useSession, signIn, signOut } from "next-auth/react";
+import { useEffect } from "react";
+import styled from "styled-components";
+import Link from "next/link";
+
+const IconLogIn = styled.span`
+  color: ${({ theme }) => theme.cardPrimary};
+  border: 1px solid ${({ theme }) => theme.text};
+  border-radius: 50%;
+  width: 40px;
+  height: 40px;
+`;
+
+const IconLogOut = styled.span`
+  color: ${({ theme }) => theme.cardPrimary};
+  border: 1px solid ${({ theme }) => theme.text};
+  border-radius: 50%;
+  width: 40px;
+  height: 40px;
+  & img {
+    filter: ${({ theme }) =>
+      theme.navbarText === "#a3a8c8"
+        ? "invert(0.6) brightness(1) sepia(0.5) hue-rotate(210deg) saturate(1) contrast(1)"
+        : "brightness(0)"};
+  }
+`;
+
+const StyledButton = styled.button`
+  height: 20px;
+  background-color: ${({ theme }) => theme.cardPrimary};
+  color: ${({ theme }) => theme.cardPrimaryText};
+  cursor: pointer;
+`;
+
+export default function Login({
+  handleCreateUser,
+  handleCheckUserExistence,
+  handleToggleThemeMode,
+  handlePickupUserThemeMode,
+}) {
+  const { data: session } = useSession();
+
+  useEffect(() => {
+    async function checkUserExistence() {
+      if (session) {
+        const userId = session.user.id;
+        let userIsAvailable = (await handleCheckUserExistence({ userId }))
+          ? true
+          : false;
+        let currentUserThemeMode = await handlePickupUserThemeMode({ userId });
+        handleToggleThemeMode(currentUserThemeMode);
+        if (!userIsAvailable) {
+          const userId = session.user.id;
+          handleCreateUser({ userId });
+        }
+      }
```

Comment:
This is only a coding style suggestion so feel free to ignore. When you have multiple nested if statements, the code becomes hard to read. You can instead try to reduce the number of nestings whenever possible by using early return statements:
```suggestion
      if (!session) { 
        return; 
      }
      
      const userId = session.user.id;

      const currentUserThemeMode = await handlePickupUserThemeMode({ userId });
      handleToggleThemeMode(currentUserThemeMode);
      
      const userIsAvailable = await handleCheckUserExistence({ userId })
      if (!userIsAvailable) {
        handleCreateUser({ userId });
      }
```

Reply:
suggestions adopted